### PR TITLE
Thread density

### PR DESCRIPTION
**Pull request by LunarX**
There are a few other features in this PR like last one

Depends on #224

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt [hunk 1]
```diff
@@ -397,4 +397,10 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
     }
 
     private fun scrollToTop() = binding.threadsList.layoutManager?.scrollToPosition(0)
+
+    enum class ThreadDensity {
+        COMPACT,
+        NORMAL,
+        LARGE
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt:404]:
```suggestion
        LARGE,
```

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt [hunk 2]
```diff
@@ -97,22 +93,22 @@ fun View.toggleChevron(
...
 }
 
-fun View.setMargins(left: Int = 0, top: Int = 0, right: Int = 0, bottom: Int = 0) {
+fun View.setMargins(left: Int? = null, top: Int? = null, right: Int? = null, bottom: Int? = null) {
     if (layoutParams is ViewGroup.MarginLayoutParams) {
-        (layoutParams as ViewGroup.MarginLayoutParams).setMargins(left, top, right, bottom)
+        (layoutParams as ViewGroup.MarginLayoutParams).apply {
+            setMargins(
+                left ?: leftMargin,
+                top ?: leftMargin,
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:101]:
```suggestion
                top ?: topMargin,
```

File: app/src/main/java/com/infomaniak/mail/views/AvatarView.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import com.infomaniak.lib.core.utils.firstOrEmpty
+import com.infomaniak.lib.core.utils.loadAvatar
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.models.Correspondent
+import com.infomaniak.mail.databinding.ViewAvatarBinding
+
+class AvatarView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/views/AvatarView.kt:36]:
```suggestion
    defStyleAttr: Int = 0,
```

File: app/src/main/java/com/infomaniak/mail/utils/Extensions.kt [hunk 4]
```diff
@@ -97,22 +93,22 @@ fun View.toggleChevron(
...
+                right ?: rightMargin,
+                bottom ?: bottomMargin
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/Extensions.kt:103]:
```suggestion
                bottom ?: bottomMargin,
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by JorisBodin — CHANGES_REQUESTED** [review]
Waiting for changes

**Review by JorisBodin — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**